contrib/inventory/azure_rm.py

```diff
@@ -646,11 +647,15 @@ def _load_machines(self, machines):
                     for ip_config in network_interface.ip_configurations:
                         host_vars['private_ip'] = ip_config.private_ip_address
                         host_vars['private_ip_alloc_method'] = ip_config.private_ip_allocation_method
+                        if self.include_private_ip:
+                            host_vars['ansible_host'] = ip_config.private_ip_address
                         if ip_config.public_ip_address:
                             public_ip_reference = self._parse_ref_id(ip_config.public_ip_address.id)
                             public_ip_address = self._network_client.public_ip_addresses.get(
                                 public_ip_reference['resourceGroups'],
                                 public_ip_reference['publicIPAddresses'])
+                            if not self.include_private_ip:
+                                host_vars['ansible_host'] = public_ip_address.ip_address
                             host_vars['ansible_host'] = public_ip_address.ip_address
```

Comment:
Sorry, I m really confused what's the logic of the `include_private_ip`. Is this 
```
ANSIBLE[HOST] = INCLUDE_PRIVATE_IP ? PRIVATE_IP : PUBLIC_IP
```
or something else?

Reply:
If private_ips are not enabled then return public ip addresses for the ansible_host. This maintains the original data that is returned.

Reply:
Seems the line after the conditional check is a potential bug..

Reply:
Why is that? That's what the line was before just with an if statement in case private ips are set?

Reply:
My point is from [line 657 to 659](https://github.com/rossedman/ansible/blob/b72d0a46218baf2a98df19e703ce8647b65b5d6a/contrib/inventory/azure_rm.py#L657-L659), you always set `host_vars['ansible_host'] = public_ip_address.ip_address`, no matter what the conditional check result is

Reply:
Oh, man. That's not supposed to be there. I see what you mean. I had run this for awhile and then ported over. Let me push update. Duh.